libraries/lib-wave-track/CachingPlayableSequence.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "AudioIOSequences.h"
+#include "AudioSegmentSampleView.h"
+
+#include <memory>
+
+class WaveTrack;
+
+class WAVE_TRACK_API CachingPlayableSequence : public PlayableSequence
+{
+public:
+   CachingPlayableSequence(const WaveTrack&);
+
+   // WideSampleSequence
+   size_t NChannels() const override;
+   float GetChannelGain(int channel) const override;
+   size_t GetMaxBlockSize() const override;
+   sampleCount GetBlockStart(sampleCount t) const override;
+   size_t GetBestBlockSize(sampleCount t) const override;
+   bool Get(
+      size_t iChannel, size_t nBuffers, samplePtr buffers[],
+      sampleFormat format, sampleCount start, size_t len,
+      fillFormat fill = fillZero, bool mayThrow = true,
+      sampleCount* pNumWithinClips = nullptr) const override;
+   double GetStartTime() const override;
+   double GetEndTime() const override;
+   double GetRate() const override;
+   sampleFormat WidestEffectiveFormat() const override;
+   bool HasTrivialEnvelope() const override;
+   void GetEnvelopeValues(
+      double* buffer, size_t bufferLen, double t0) const override;
+
+   AudioGraph::ChannelType GetChannelType() const override;
+
+   // PlayableSequence
+   bool IsLeader() const override;
+   bool GetSolo() const override;
+   bool GetMute() const override;
+
+private:
+   const WaveTrack& mWaveTrack;
+   // Ok to have it mutable so long as it is used by one thread only.
+   // One per channel, one per track.
```

Comment:
I think the comment should simply say "One per channel"

Reply:
Thank you.